MainRepository/Program.cs

```diff
@@ -1,7 +1,55 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
-}
\ No newline at end of file
+﻿// This program validates the file path
+// BNF diagram of finding valid file path
+using static State;
+var strs = new List<string> { @"C:\PROGRAMS\DATA\README.TXT", @"C:\PROGRAMS\README.TXT"
+   , @"C:\PROGRAMS\DATA\DIR\README.TXT", @"C:\README.TXT", "..", "", "C:"
+   , "C:\\PROGRAMS\\README..TXT" };
+foreach (var str in strs) {
+   if (IsValidFilePath (str)) {
+      Console.ForegroundColor = ConsoleColor.Green;
+      PrintValidPath (str);
+      var (dir, folder, filename, ext) = Result (str);
+      Console.WriteLine ($"{dir}, {folder}, {filename}, {ext}");
+   } else {
+      Console.ForegroundColor = ConsoleColor.Red;
+      PrintValidPath (str);
+   }
+}
+
+// Extracts drive, directory, filename and extension from the given file path 
+(string dir, string folder, string filename, string ext) Result (string input) {
+   string drive = Path.GetPathRoot (input)!;
+   drive = drive.Replace ("\\", "");
+   string folders = Path.GetDirectoryName (input)!.Substring (drive.Length);
+   string fileName = Path.GetFileNameWithoutExtension (input);
+   string extension = Path.GetExtension (input).TrimStart ('.');
+   return (drive, folders, fileName, extension);
+}
+
+// Checks if the given string is valid filepath or not
+bool IsValidFilePath (string input) {
+   State s = A;
+   foreach (var ch in input.Trim ().ToUpper () + "~") {
+      s = (s, ch) switch {
+         (A, >= 'A' and <= 'Z') => B,
+         (B, ':') => C,
+         (C, '\\') => D,
+         (D, >= 'A' and <= 'Z') => D,
+         (D, '\\') => E,
+         (E or F, >= 'A' and <= 'Z') => F,
+         (F, '\\') => F,
+         (F, '.') => G,
+         (G or H, >= 'A' and <= 'Z') => H,
+         (H, '~') => I,
+         _ => Z,
+      };
+   }
+   return (s == I);
+}
+
+void PrintValidPath (string path) {

```

Comment:
This method actually prints both valid and invalid path. So the name is incorrect.
Moreover, I expected this to take the color also as a parameter to set the foreground color.
```suggestion
void PrintPath (string path, ConsoleColor clr) {

```